mypy/subtypes.py

```diff
@@ -435,8 +435,10 @@ def visit_instance(self, left: Instance) -> bool:
             # dynamic base classes correctly, see #5456.
             return not isinstance(self.right, NoneType)
         right = self.right
-        if isinstance(right, TupleType) and right.partial_fallback.type.is_enum:
-            return self._is_subtype(left, mypy.typeops.tuple_fallback(right))
+        if isinstance(right, TupleType):
+            return self._is_subtype(left, right.partial_fallback) and self._is_subtype(
+                left, mypy.typeops.tuple_fallback(right)
+            )
```

Comment:
Sorry, but this is wrong. IIUC this will make `tuple[object, ...]` a subtype of `tuple[int, str]`. While even `tuple[int, ...]` shoul _not_ be a subtype of `tuple[int, int]`. Really only `tuple[Any, ...]` should be a subtype of say `tuple[int, int]`. This is the same as we consider `*args: Any` in callables very differently from `*args: object` or `*args: int`.

Note btw this question will appear even broader in context of my upcoming PR for variadic types, e.g. should we also special case `Foo[int, *tuple[Any, ...], str]` similarly in spirit to extending the special-casing of trivial arguments to trivial suffix in https://github.com/python/mypy/pull/15913. (Btw I would say yes, but it should be a separate PR after we gain more experience).

Also I bet most of the unused type ignores in mypy primer are from situations like:
```python
v: tuple[int, ...]
if len(v) == 2:
    x: tuple[int, int] = v
```
(as e.g. for ignore in mypy itself). See also https://github.com/python/mypy/issues/1178. The proper way to solve such issues is by finally completing https://github.com/python/mypy/pull/10367. Unfortunately author is not responding anymore, so one of us may need to complete the PR, since it is an important one.